extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/radio/FrontierSiliconRadioConnection.java

```diff
@@ -0,0 +1,208 @@
+
+    /** Access pin, passed upon login as GET parameter. */
+    private final String pin;
+
+    /** The session ID we get from the radio after logging in. */
+    private String sessionId;
+
+    /** http clients, store cookies, so it is kept in connection class. */
+    private HttpClient httpClient = null;
+
+    /** Flag indicating if we are successfully logged in. */
+    private boolean isLoggedIn = false;
+
+    public FrontierSiliconRadioConnection(String hostname, int port, String pin) {
+        this.hostname = hostname;
+        this.port = port;
+        this.pin = pin;
+    }
+
+    /**
+     * Perform login/establish a new session. Uses the PIN number and when successful saves the assigned sessionID for
+     * future requests.
+     *
+     * @return <code>true</code> if login was successful; <code>false</code> otherwise.
+     * @throws IOException if communication with the radio failed, e.g. because the device is not reachable.
+     */
+    public boolean doLogin() throws IOException {
+        isLoggedIn = false; // reset login flag
+
+        if (httpClient == null) {
+            httpClient = new HttpClient();
+        }
+
+        final String url = "http://" + hostname + ":" + port + "/fsapi/CREATE_SESSION?pin=" + pin;
+
+        logger.trace("opening URL:" + url);
+
+        final HttpMethod method = new GetMethod(url);
+        method.getParams().setSoTimeout(SOCKET_TIMEOUT);
+        method.getParams().setParameter(HttpMethodParams.RETRY_HANDLER, new DefaultHttpMethodRetryHandler(3, false));
+
+        try {
+            final int statusCode = httpClient.executeMethod(method);
+            if (statusCode != HttpStatus.SC_OK) {
+                logger.warn("Method failed: " + method.getStatusLine());
+                if (method.getStatusCode() == 403) {
+                    // forbidden! maybe wrong pin?
+                    throw new RuntimeException("Radio does not allow connection, maybe wrong pin?");
+                }
+            }
+
+            final String responseBody = IOUtils.toString(method.getResponseBodyAsStream());
+            if (!responseBody.isEmpty()) {
+                logger.trace("login response: " + responseBody);
+            }
+
+            try {
+                final FrontierSiliconRadioApiResult result = new FrontierSiliconRadioApiResult(responseBody);
+                if (result.isStatusOk()) {
+                    logger.trace("login successful");
+                    sessionId = result.getSessionId();
+                    isLoggedIn = true;
+                    return true; // login successful :-)
+                }
+            } catch (Exception e) {
+                logger.error("Parsing response failed");
+            }
+
+        } catch (HttpException he) {
+            logger.error("Fatal protocol violation: {}", he.toString());
+            throw he;
+        } catch (IOException ioe) {
+            logger.error("Fatal transport error: {}", ioe.toString());
```

Comment:
reduce to debug - this is actually an expected exception (like UnknownHostException), in which case we will anyhow set the thing to offline and provide the exception message as a detail